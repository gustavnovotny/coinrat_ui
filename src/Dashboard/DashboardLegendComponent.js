// @flow
import React from "react"
import {STATUS_CANCELED, STATUS_CLOSED, STATUS_OPEN} from "../Orders/Order"
import {ORDERS_STATUS_COLORS} from "../Orders/ChartColors"
import HelpIconComponent from "../HelpIconComponent"

type Props = {
  isSimulationModeOn: boolean
}

const DashboardLegendComponent = ({isSimulationModeOn}: Props) => {
  const statuses = ORDERS_STATUS_COLORS

  const mockMarketTooltip = <div style={{maxWidth: 256 + 'px'}}>
    <p>Mock market is virtual market that simulates behaviour of stock-market.</p>
    <p>All orders are processed as <strong>closed</strong> immediately.</p>
  </div>

  const simulationLegend = isSimulationModeOn ? <div><p>
      Strategy executed from UI is <strong>always</strong> executed against
      {' '}
      <strong>mock market</strong>
      {' '}
      <HelpIconComponent helpText={mockMarketTooltip}/>
      .
    </p>
      <p>Executing strategy against real market is possible only via command line on the backend server.</p>
    </div>
    : ''

  return <div>
    {simulationLegend}
    <h5>Legend</h5>
    <ul className="pt-list-unstyled">
      <li><span style={{color: statuses[STATUS_OPEN]}} className="pt-icon-full-circle"/> Open order</li>
      <li><span style={{color: statuses[STATUS_CLOSED]}} className="pt-icon-full-circle"/> Closed order</li>
      <li><span style={{color: statuses[STATUS_CANCELED]}} className="pt-icon-full-circle"/> Canceled order</li>
    </ul>
  </div>
}

export default DashboardLegendComponent
