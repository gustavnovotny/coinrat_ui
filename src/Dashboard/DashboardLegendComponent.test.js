// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import DashboardLegendComponent from "./DashboardLegendComponent"

it('renders correctly when simulation mode is ON', () => {
  const tree = renderer.create(<DashboardLegendComponent isSimulationModeOn={true}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly when simulation mode is OFF', () => {
  const tree = renderer.create(<DashboardLegendComponent isSimulationModeOn={false}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
