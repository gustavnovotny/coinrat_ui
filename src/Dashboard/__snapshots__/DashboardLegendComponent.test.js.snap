// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when simulation mode is OFF 1`] = `
<div>
  
  <h5>
    Legend
  </h5>
  <ul
    className="pt-list-unstyled"
  >
    <li>
      <span
        className="pt-icon-full-circle"
        style={
          Object {
            "color": "orange",
          }
        }
      />
       Open order
    </li>
    <li>
      <span
        className="pt-icon-full-circle"
        style={
          Object {
            "color": "blue",
          }
        }
      />
       Closed order
    </li>
    <li>
      <span
        className="pt-icon-full-circle"
        style={
          Object {
            "color": "black",
          }
        }
      />
       Canceled order
    </li>
  </ul>
</div>
`;

exports[`renders correctly when simulation mode is ON 1`] = `
<div>
  <div>
    <p>
      Strategy executed from UI is 
      <strong>
        always
      </strong>
       executed against
       
      <strong>
        mock market
      </strong>
       
      <span>
        <div
          className="pt-popover-target"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span
            className="pt-icon-standard pt-icon-info-sign pt-intent-primary"
            disabled={undefined}
            tabIndex={0}
          />
        </div>
      </span>
      .
    </p>
    <p>
      Executing strategy against real market is possible only via command line on the backend server.
    </p>
  </div>
  <h5>
    Legend
  </h5>
  <ul
    className="pt-list-unstyled"
  >
    <li>
      <span
        className="pt-icon-full-circle"
        style={
          Object {
            "color": "orange",
          }
        }
      />
       Open order
    </li>
    <li>
      <span
        className="pt-icon-full-circle"
        style={
          Object {
            "color": "blue",
          }
        }
      />
       Closed order
    </li>
    <li>
      <span
        className="pt-icon-full-circle"
        style={
          Object {
            "color": "black",
          }
        }
      />
       Canceled order
    </li>
  </ul>
</div>
`;
